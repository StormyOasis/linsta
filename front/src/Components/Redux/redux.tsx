import { configureStore } from '@reduxjs/toolkit';
import { useDispatch, useSelector } from 'react-redux';

import authSliceCreator from './slices/auth.slice';
import modalSliceCreator from './slices/modals.slice';
import profileSliceCreator from './slices/profile.slice';

export type Actions = {
    authActions?: any,
    modalActions?: any;
    profileActions?: any;
}

export const actions: Actions = {};

export const buildStore = (initialState?: any) => {
    const preloadedState: any = initialState == null ? {} : initialState;
    const { authActions, authReducer } = authSliceCreator(preloadedState.auth);
    const { modalActions, modalReducer } = modalSliceCreator(preloadedState.modal);
    const { profileActions, profileReducer } = profileSliceCreator(preloadedState.profile);

    actions.authActions = authActions;
    actions.modalActions = modalActions;
    actions.profileActions = profileActions;

    const store = configureStore({
        devTools: true,
        reducer: {
            auth: authReducer,
            modal: modalReducer,
            profile: profileReducer
        },
        ...preloadedState
    });

    return store;
}

export type ConfiguredStore = ReturnType<typeof buildStore>;
export type StoreGetState = ConfiguredStore["getState"];
export type RootState = ReturnType<StoreGetState>;
export type AppDispatch = ConfiguredStore["dispatch"];

// Use throughout instead of `useDispatch` and `useSelector`
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();