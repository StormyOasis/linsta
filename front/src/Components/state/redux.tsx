import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "./reducers/rootReducer";
import { initialState } from "./store";

const store = configureStore({
    reducer: rootReducer,
    devTools: true,
    preloadedState: initialState,
  });
  
  export type AppDispatch = typeof store.dispatch;
  export type RootState = ReturnType<typeof store.getState>;
  
  export default store;
  