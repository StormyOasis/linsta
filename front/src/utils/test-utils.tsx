import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RootState } from "../Components/Redux/redux";
import { authInitialState } from '../Components/Redux/slices/auth.slice';
import { modalInitialState } from '../Components/Redux/slices/modals.slice';
import { profileInitialState } from '../Components/Redux/slices/profile.slice';
import { miscInitialState } from '../Components/Redux/slices/misc.slice';
import { configureStore } from '@reduxjs/toolkit';

import Theme from '../Components/Themes/Theme';

export const mockInitialState = (overrides?: Partial<RootState>): RootState => ({
    auth: { ...authInitialState, ...(overrides?.auth ?? {}) },
    modal: { ...modalInitialState, ...(overrides?.modal ?? {}) },
    profile: { ...profileInitialState, ...(overrides?.profile ?? {}) },
    misc: { ...miscInitialState, ...(overrides?.misc ?? {}) },
});

type ExtendedRenderOptions = {
    store?: ReturnType<typeof configureStore>;
    preloadedState?: any;
    route?: string; // ‚Üê allow overriding the initial route
} & Omit<RenderOptions, "queries">;

const dummyAuthReducer = (state = { user: null, isLoading: false, error: null }, _action: any) => state;
const dummyModalReducer = (state = { isOpen: false }, _action: any) => state;
const dummyProfileReducer = (state = { name: '', email: '' }, _action: any) => state;
const dummyMiscReducer = (state = { someData: null }, _action: any) => state;

const renderWithStore = (
    ui: ReactElement,
    {
        preloadedState,
        store = configureStore({
            reducer: {
                auth: dummyAuthReducer,
                modal: dummyModalReducer,
                profile: dummyProfileReducer,
                misc: dummyMiscReducer,
            },
            preloadedState,
        }),
        route = "/",
        ...renderOptions
    }: ExtendedRenderOptions = {}
) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[route]}>
                    <Theme>{children}</Theme>
                </MemoryRouter>
            </Provider>
        );
    };

    return {
        ...render(ui, { wrapper: Wrapper, ...renderOptions }),
        store,
    };
};

export * from "@testing-library/react";
export { renderWithStore };
export type { ExtendedRenderOptions };