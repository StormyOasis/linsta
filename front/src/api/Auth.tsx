import { postLogin } from "./ServiceController";

export type AuthUser = {
    token: string;
    id: string;
    userName: string;
}

export const authHeader = () => {
    const headers:any = {};

    // Authentication and Authorization via JWT
    const user = JSON.parse(localStorage.getItem('user') as string) as AuthUser;
    if (user == null) {
        return headers;
    }

    if (user && user.token) {
        headers['Authorization'] = `Bearer ${user.token}`;
    }

    return headers;
}

export const login = async (userName: string, password: string) => {    
    if (userName.trim().length === 0 || password.trim().length === 0) {
        throw new Error("Invalid username or password");
    }

    const result = await postLogin({ userName, password });

    if (result == null || result.status != 200 || result.data == null || result.data.token == null) {
        throw new Error("Invalid username or password");
    }

    return result.data;
}

export const logout = () => {
    localStorage.removeItem("user");
    localStorage.removeItem("recentSearches");
}

export const getCurrentUser = () => {
    const user = localStorage.getItem('user');
    if (user == null) {
        return null;
    }
    return JSON.parse(user);
}