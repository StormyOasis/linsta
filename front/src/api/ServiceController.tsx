import axios from "axios";
import { authHeader } from "./Auth";

let host = "http://localhost:3001"; //TODO: From config or env

export type ServiceResponse = {
    data: any;
    status: number;
    statusText: string;
}

export const postAccountsAttempt = async (data: any): Promise<ServiceResponse> => {
    const res = await axios.post(`${host}/api/v1/accounts/attempt/`, data);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText,
    }
};

export const getAccountsCheckUserUnique = async (value: string | number): Promise<ServiceResponse> => {
    const res = await axios.get(`${host}/api/v1/accounts/check/${value}`);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText
    }
};

export const getAccountsSendVerifyNotification = async (userData: string): Promise<ServiceResponse>  => {
    const res = await axios.get(`${host}/api/v1/accounts/send_confirm_code?user=${userData}`);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText
    }
}

export const postLogin = async (data: any): Promise<ServiceResponse> => {
    const res = await axios.post(`${host}/api/v1/accounts/login`, data);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText,
    }
}

export const postForgotPassword = async (data: any): Promise<ServiceResponse> => {
    const res = await axios.post(`${host}/api/v1/accounts/forgot`, data);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText,
    }
}

export const postChangePassword = async (data: any): Promise<ServiceResponse> => {
    const res = await axios.post(`${host}/api/v1/accounts/change_password`, data);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText,
    }
}