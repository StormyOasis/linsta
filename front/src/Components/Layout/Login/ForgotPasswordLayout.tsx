import React, { useState } from "react";
import { styled } from "styled-components";

import * as styles from "/src/Components/Common/CombinedStyling";

import Theme from "/src/Components/Themes/Theme";
import StyledLink from "/src/Components/Common/StyledLink";
import StyledInput from "/src/Components/Common/StyledInput";
import StyledButton from "/src/Components/Common/StyledButton";
import Modal from "/src/Components/Common/Modal";
import { postForgotPassword } from "/src/api/ServiceController";

type ForgotProps = {

};

const ForgotLayoutWrapper = styled.main<any>`
  display: flex;
  flex-direction: column;
  flex-grow: 1;
`;

const ForgotHeaderImage = styled.span`
    background-image: url('/public/images/forgot.svg');
    display: block; 
    height: 133px; 
    width: 133px;
`;

const HeadingText = styled.div`
    margin-top: 10px;
    text-align:center;
    font-weight: 600;        
`;

const SubHeadingText = styled.div`
    color: rgb(115,115,155);
    text-align: center;
    margin-top:6px;
    margin-bottom: 10px;
`;

const ForgotForm = styled.form`
  display: flex;
  flex-direction: column;
  margin: 0;
  max-width: 350px;
  vertical-align: baseline;
`;

const SignupText = styled.div`
    margin-top: 10px;
    margin-bottom: 10px;
    text-align: center;
`;

const ModalContentWrapper = styled.div`
    align-content: stretch;
    align-items: center;
    border: none;
    display: flex;
    flex-direction: column;
    flex-grow: 0;
    flex-shrink: 0;
    justify-content: flex-start;
    overflow: visible;
    pointer-events: all;
    position: relative;
    margin: 20px 28px 20px 28px;
`;

const ModalSectionWrapper = styled.div`
    align-content: stretch;
    align-items: stretch;
    display: flex;
    flex-direction: column;
    flex-grow: 0;
    flex-shrink: 0;
    justify-content: flex-start;
    overflow: visible;
    position: relative;
    pointer-events: all;
`;

const renderModal = (showModal, setShowModal, queryResponse) => {
    if(queryResponse == null) {
        throw new Error("Invalid query data");
    }

    const cont = document.getElementById("modalContainer");
    const sectionCont = document.getElementById("mainSectionContainer");
    if (cont && sectionCont) {
        cont.style.height = "100%";
        sectionCont.style.pointerEvents = "none";
    }

    return (
        <>
            <Modal title="Forgot Password" onClose={()=> {
                setShowModal(false);
                if (cont && sectionCont) {
                    cont.style.height = "0%";
                    sectionCont.style.pointerEvents = "auto";
                }                
            }}>
                <ModalContentWrapper>
                    <ModalSectionWrapper style={{ maxWidth: "350px", padding: "4px" }}>
                        <span style={{fontSize: "18px", fontWeight: 500, paddingBottom: "20px", textAlign: "center"}}>{queryResponse.title}</span>
                        <div style={{fontWeight: 400, paddingBottom: "10px", textAlign: "center"}}>
                            {queryResponse.text}
                        </div>
                    </ModalSectionWrapper>        
                </ModalContentWrapper>
            </Modal>
        </>
    );
}

const onSubmit = async (userData: string, setShowModal, setQueryResponse) => {
    try {
        const result = await postForgotPassword({user: userData});

        if(result.status == 200) {
            setQueryResponse({title: result.data.title, text: result.data.text});
            setShowModal(true);
        }
        else {
            throw new Error();
        }

    } catch(err) {
        console.error(err);
        return false;
    }
    return true;
}

const renderForgotForm = () => {
    const [userData, setUserData] = useState('');
    const [showModal, setShowModal] = useState(false);
    const [queryResponse, setQueryResponse] = useState(null);

    return (
        <>
            {showModal && renderModal(showModal, setShowModal, queryResponse)}
            <ForgotHeaderImage/>
            <ForgotForm method="post">
                <HeadingText>Trouble logging in?</HeadingText>
                <SubHeadingText>
                    Enter your email, phone, or username and we'll send you a link to get back into your account.
                </SubHeadingText> 
                <StyledInput
                    name="userName"
                    placeholder="Email, Phone, or Username"
                    value={userData}
                    isValid={userData.length > 0}
                    onChange={(event: React.ChangeEvent<HTMLInputElement>) => setUserData(event.target.value)} />
                <StyledButton
                    type="button"
                    disabled={userData.length == 0}
                    onClick={() => {return onSubmit(userData, setShowModal, setQueryResponse);}}
                    text="Send Login Link">
                </StyledButton>
                <div className={styles.default.signupFormDiv1}>
                    <div className={styles.default.signupFormDiv2}>
                        <div className={styles.default.signupFormDiv3} />
                        <div className={styles.default.signupFormDiv4}>OR</div>
                        <div className={styles.default.signupFormDiv3} />
                    </div>
                </div>
                <SignupText>
                    Don't have an account? <StyledLink to="/signup">Sign up</StyledLink>                    
                </SignupText>  
            </ForgotForm>           
        </>
    );
}

const ForgotPasswordLayout: React.FC<ForgotProps> = (props: ForgotProps) => {
    return (
        <Theme>
            <ForgotLayoutWrapper role="main">
                <div className={styles.default.innerDiv1}>
                    <div className={styles.default.innerDiv2}>
                        <div className={styles.default.signupBox}>
                            {renderForgotForm()}
                        </div>
                    </div>
                </div>
            </ForgotLayoutWrapper>
        </Theme>
    );
}

export default ForgotPasswordLayout;