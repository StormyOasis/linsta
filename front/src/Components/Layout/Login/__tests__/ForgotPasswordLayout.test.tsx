import React from 'react'
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import ForgotPasswordLayout from '../ForgotPasswordLayout';

import { renderWithStore, mockInitialState } from '../../../../utils/test-utils';

import * as serviceController from '../../../../api/ServiceController';
import { actions, buildStore } from '../../../Redux/redux';
import { MODAL_TYPES } from '../../../Redux/slices/modals.slice';

jest.mock('../../../../api/ServiceController');


describe("ForgotPasswordLayout", () => {
    let store: ReturnType<typeof buildStore>;

    beforeAll(() => {
        store = buildStore(); // âœ… Save the store that initializes actions.modalActions
    });

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("dispatches openModal with success response on clicking send link", async () => {
        const fakeResponse = {
            status: 200,
            data: {
                title: "Success!",
                text: "A login link has been sent to your email.",
            },
        };
        (serviceController.postForgotPassword as jest.Mock).mockResolvedValueOnce(
            fakeResponse
        );

        // Spy on openModal modal action
        const openModalSpy = jest.spyOn(actions.modalActions, "openModal");

        const r = renderWithStore(<ForgotPasswordLayout />, { store })
        const input = screen.getByPlaceholderText(/Email, Phone, or Username/i);
        const sendButton = screen.getByRole("button", { name: /Send Login Link/i });

        await userEvent.type(input, "user@example.com");
        userEvent.click(sendButton);

        await waitFor(() => {
            expect(openModalSpy).toHaveBeenCalled();
        });
        expect(r).toMatchSnapshot();
    });

    it("dispatches openModal with error response on API failure", async () => {
        (serviceController.postForgotPassword as jest.Mock).mockRejectedValueOnce({
            response: {
                status: 400,
                data: { message: "Bad request" }
            }
        })

        // Spy on openModal modal action
        const openModalSpy = jest.spyOn(actions.modalActions, "openModal");

        const r = renderWithStore(<ForgotPasswordLayout />, { store });
        const input = screen.getByPlaceholderText(/Email, Phone, or Username/i);
        const sendButton = screen.getByRole("button", { name: /Send Login Link/i });

        await userEvent.type(input, "user@example.com");
        userEvent.click(sendButton);

        await waitFor(() => {
            expect(openModalSpy).toHaveBeenCalledWith({
                modalName: MODAL_TYPES.FORGOT_PASSWORD_MODAL,
                data: {
                    queryResponseTitle: "Error Sending Link",
                    queryResponseText:
                        "An unexpected error has occurred while sending the login link",
                },
            });
        });
        expect(r).toMatchSnapshot();
    });
});