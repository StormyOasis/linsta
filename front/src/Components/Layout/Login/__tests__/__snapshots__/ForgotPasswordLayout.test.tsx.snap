// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordLayout dispatches openModal with error response on API failure 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="ForgotPasswordLayout__ForgotLayoutWrapper-sc-upavat-0 cHylCf"
        role="main"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
            >
              <span
                class="CombinedStyling__Span-sc-c3jdxk-1 ForgotPasswordLayout__ForgotHeaderImage-sc-upavat-1 kSRMWv kLOkNx"
              />
              <form
                class="ForgotPasswordLayout__ForgotForm-sc-upavat-4 hREDts"
                method="post"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__HeadingText-sc-upavat-2 escqfF fRNugz"
                >
                  Trouble logging in?
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__SubHeadingText-sc-upavat-3 escqfF bXJVdK"
                >
                  Enter your email, phone, or username and we'll send you a link to get back into your account.
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
                >
                  <label
                    class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
                  >
                    <input
                      class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
                      data-testid="userName"
                      maxlength="255"
                      placeholder="Email, Phone, or Username"
                      value="user@example.com"
                    />
                    <div
                      class="CombinedStyling__Div-sc-c3jdxk-0 StyledInput__ValidationIcon-sc-fkf055-3 hNEsIT kYVJfy"
                    >
                      <svg
                        aria-hidden="true"
                        class="Icon-sc-btq2kp-0"
                        fill="currentColor"
                        focusable="false"
                        name="check-circle"
                        stroke="none"
                        stroke-width="2"
                      >
                        <use
                          href="http://localhost:8080/public/images/sprite.svg#check-circle"
                        />
                      </svg>
                    </div>
                  </label>
                </div>
                <button
                  class="StyledButton__StyledButtonComponent-sc-8t4kac-0 bZbOLt"
                  type="button"
                >
                  Send Login Link
                </button>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                >
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                  >
                    <div
                      class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                    />
                    <div
                      class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                    >
                      OR
                    </div>
                    <div
                      class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                    />
                  </div>
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__SignupText-sc-upavat-5 escqfF jCpmzb"
                >
                  Don't have an account? 
                  <a
                    class="CombinedStyling__Link-sc-c3jdxk-7 StyledLink__StyledLinkWrapperFromLink-sc-1rgr01q-0 rKNtR dnJNFP"
                    href="/signup"
                    role=""
                  >
                    Sign up
                  </a>
                </div>
              </form>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="ForgotPasswordLayout__ForgotLayoutWrapper-sc-upavat-0 cHylCf"
      role="main"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <span
              class="CombinedStyling__Span-sc-c3jdxk-1 ForgotPasswordLayout__ForgotHeaderImage-sc-upavat-1 kSRMWv kLOkNx"
            />
            <form
              class="ForgotPasswordLayout__ForgotForm-sc-upavat-4 hREDts"
              method="post"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__HeadingText-sc-upavat-2 escqfF fRNugz"
              >
                Trouble logging in?
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__SubHeadingText-sc-upavat-3 escqfF bXJVdK"
              >
                Enter your email, phone, or username and we'll send you a link to get back into your account.
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
              >
                <label
                  class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
                >
                  <input
                    class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
                    data-testid="userName"
                    maxlength="255"
                    placeholder="Email, Phone, or Username"
                    value="user@example.com"
                  />
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 StyledInput__ValidationIcon-sc-fkf055-3 hNEsIT kYVJfy"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon-sc-btq2kp-0"
                      fill="currentColor"
                      focusable="false"
                      name="check-circle"
                      stroke="none"
                      stroke-width="2"
                    >
                      <use
                        href="http://localhost:8080/public/images/sprite.svg#check-circle"
                      />
                    </svg>
                  </div>
                </label>
              </div>
              <button
                class="StyledButton__StyledButtonComponent-sc-8t4kac-0 bZbOLt"
                type="button"
              >
                Send Login Link
              </button>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                >
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                  />
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                  >
                    OR
                  </div>
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                  />
                </div>
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__SignupText-sc-upavat-5 escqfF jCpmzb"
              >
                Don't have an account? 
                <a
                  class="CombinedStyling__Link-sc-c3jdxk-7 StyledLink__StyledLinkWrapperFromLink-sc-1rgr01q-0 rKNtR dnJNFP"
                  href="/signup"
                  role=""
                >
                  Sign up
                </a>
              </div>
            </form>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ForgotPasswordLayout dispatches openModal with success response on clicking send link 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="ForgotPasswordLayout__ForgotLayoutWrapper-sc-upavat-0 cHylCf"
        role="main"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
            >
              <span
                class="CombinedStyling__Span-sc-c3jdxk-1 ForgotPasswordLayout__ForgotHeaderImage-sc-upavat-1 kSRMWv kLOkNx"
              />
              <form
                class="ForgotPasswordLayout__ForgotForm-sc-upavat-4 hREDts"
                method="post"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__HeadingText-sc-upavat-2 escqfF fRNugz"
                >
                  Trouble logging in?
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__SubHeadingText-sc-upavat-3 escqfF bXJVdK"
                >
                  Enter your email, phone, or username and we'll send you a link to get back into your account.
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
                >
                  <label
                    class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
                  >
                    <input
                      class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
                      data-testid="userName"
                      maxlength="255"
                      placeholder="Email, Phone, or Username"
                      value="user@example.com"
                    />
                    <div
                      class="CombinedStyling__Div-sc-c3jdxk-0 StyledInput__ValidationIcon-sc-fkf055-3 hNEsIT kYVJfy"
                    >
                      <svg
                        aria-hidden="true"
                        class="Icon-sc-btq2kp-0"
                        fill="currentColor"
                        focusable="false"
                        name="check-circle"
                        stroke="none"
                        stroke-width="2"
                      >
                        <use
                          href="http://localhost:8080/public/images/sprite.svg#check-circle"
                        />
                      </svg>
                    </div>
                  </label>
                </div>
                <button
                  class="StyledButton__StyledButtonComponent-sc-8t4kac-0 bZbOLt"
                  type="button"
                >
                  Send Login Link
                </button>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                >
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                  >
                    <div
                      class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                    />
                    <div
                      class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                    >
                      OR
                    </div>
                    <div
                      class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                    />
                  </div>
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__SignupText-sc-upavat-5 escqfF jCpmzb"
                >
                  Don't have an account? 
                  <a
                    class="CombinedStyling__Link-sc-c3jdxk-7 StyledLink__StyledLinkWrapperFromLink-sc-1rgr01q-0 rKNtR dnJNFP"
                    href="/signup"
                    role=""
                  >
                    Sign up
                  </a>
                </div>
              </form>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="ForgotPasswordLayout__ForgotLayoutWrapper-sc-upavat-0 cHylCf"
      role="main"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <span
              class="CombinedStyling__Span-sc-c3jdxk-1 ForgotPasswordLayout__ForgotHeaderImage-sc-upavat-1 kSRMWv kLOkNx"
            />
            <form
              class="ForgotPasswordLayout__ForgotForm-sc-upavat-4 hREDts"
              method="post"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__HeadingText-sc-upavat-2 escqfF fRNugz"
              >
                Trouble logging in?
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__SubHeadingText-sc-upavat-3 escqfF bXJVdK"
              >
                Enter your email, phone, or username and we'll send you a link to get back into your account.
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 StyledInput__StyledInputWrapper-sc-fkf055-0 escqfF ckFvxY fjNDtr cyFDIt"
              >
                <label
                  class="StyledInput__StyledInputLabel-sc-fkf055-2 iHXgrc"
                >
                  <input
                    class="StyledInput__StyledInputInput-sc-fkf055-1 kkxInS"
                    data-testid="userName"
                    maxlength="255"
                    placeholder="Email, Phone, or Username"
                    value="user@example.com"
                  />
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 StyledInput__ValidationIcon-sc-fkf055-3 hNEsIT kYVJfy"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon-sc-btq2kp-0"
                      fill="currentColor"
                      focusable="false"
                      name="check-circle"
                      stroke="none"
                      stroke-width="2"
                    >
                      <use
                        href="http://localhost:8080/public/images/sprite.svg#check-circle"
                      />
                    </svg>
                  </div>
                </label>
              </div>
              <button
                class="StyledButton__StyledButtonComponent-sc-8t4kac-0 bZbOLt"
                type="button"
              >
                Send Login Link
              </button>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                >
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                  />
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                  >
                    OR
                  </div>
                  <div
                    class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                  />
                </div>
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 ForgotPasswordLayout__SignupText-sc-upavat-5 escqfF jCpmzb"
              >
                Don't have an account? 
                <a
                  class="CombinedStyling__Link-sc-c3jdxk-7 StyledLink__StyledLinkWrapperFromLink-sc-1rgr01q-0 rKNtR dnJNFP"
                  href="/signup"
                  role=""
                >
                  Sign up
                </a>
              </div>
            </form>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
